File: piwigo/Login/UIComponents/PiwigoButton.m
//
//  PiwigoButton.m
//  piwigo
//
//  Created by Spencer Baker on 1/17/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//

#import "Model.h"
#import "PiwigoButton.h"

@implementation PiwigoButton

-(instancetype)init
{
	self = [super init];
	if (self)
	{
		self.layer.cornerRadius = 10.0;
		self.titleLabel.font = [UIFont piwigoFontButton];
		[self setTitleColor:[UIColor whiteColor] forState:UIControlStateNormal];
	}
	return self;
}

-(void)setHighlighted:(BOOL)highlighted
{
	[super setHighlighted:highlighted];
    
    if (AppVars.shared.isDarkPaletteActive) {
        if (highlighted) {
            self.backgroundColor = [UIColor piwigoColorOrange];
        } else {
            self.backgroundColor = [UIColor piwigoColorOrangeSelected];
        }
    } else {
        if (highlighted) {
            self.backgroundColor = [UIColor piwigoColorOrangeSelected];
        } else {
            self.backgroundColor = [UIColor piwigoColorOrange];
        }
    }
}

@end

File: piwigo/Login/UIComponents/PiwigoButton.swift
//
//  PiwigoButton.swift
//  piwigo
//
//  Created by Spencer Baker on 1/17/15.
//  Copyright (c) 2015 bakercrew. All rights reserved.
//
//  Converted to Swift 5.4 by Eddy Lelièvre-Berna on 26/02/2022.
//

import UIKit

class PiwigoButton: UIButton {
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        layer.cornerRadius = 10.0
        titleLabel?.font = UIFont.piwigoFontButton()
        setTitleColor(UIColor.white, for: .normal)
    }

    override var isHighlighted: Bool {
        get {
            return super.isHighlighted
        }
        set(highlighted) {
            super.isHighlighted = highlighted

            if AppVars.shared.isDarkPaletteActive {
                if highlighted {
                    backgroundColor = UIColor.piwigoColorOrange()
                } else {
                    backgroundColor = UIColor.piwigoColorOrangeSelected()
                }
            } else {
                if highlighted {
                    backgroundColor = UIColor.piwigoColorOrangeSelected()
                } else {
                    backgroundColor = UIColor.piwigoColorOrange()
                }
            }
        }
    }

    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
}
